File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let maxW = 0;
    let maxH = 0;

    for (const size of sizes) {
        const [w, h] = size[0] > size[1] ? size : size.reverse();
        if (w > maxW) {
            maxW = w;
        }
        if (h > maxH) {
            maxH = h;
        }
    }
    return maxW * maxH;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int maxX = 0;
        int maxY = 0;
        
        // 가로 길이를 큰 걸로 대체한다.
        for (int i=0; i<sizes.length; i++) {
            int x = Math.max(sizes[i][0], sizes[i][1]);
            int y = Math.min(sizes[i][0], sizes[i][1]);
            maxX = Math.max(x, maxX);
            maxY = Math.max(y, maxY);
        }
        
        return maxX * maxY;
    }
}